### DISCUSSION TITLE
Has anyone experienced issues with integrating Deepgram into Next.js 13?

### DISCUSSION THREAD

**Discussion by DannyPerz** [deepgram/community#237] [General help] [answered]
I'm encountering some difficulties with the integration and would appreciate any insights or solutions from the community. Thank you!"

My next js 13 program does not allow me to transcribe audio using the deepgram api, I based it on this documentation: https://deepgram.com/learn/using-next-js-for-speech-to-text

but the version of next js was updated and I can't find a solution,

My form code is this:

    if (selectedFile) {
      const formData = new FormData();
      formData.append("file", selectedFile);

      try {
        const response = await fetch("/api/transcribe", {
          method: "POST",
          body: formData,
        });
        if (response.ok) {
          const data = await response.json();
          const transcriptionText = 
          data.results.channels[0].alternatives[0].paragraphs.transcript;
          console.log(transcriptionText // error here
        } else {
          console.log("Error ", response.statusText);
        }
      } catch (error) {
        toast.error("Error ", {
          // + error
          autoClose: 3000,
          hideProgressBar: false,
          theme: "dark",
        });
        console.log(error);
      }
    } else {
      toast.warn("Not selected.", {
        autoClose: 1000,
        hideProgressBar: true,
        theme: "dark",
      });
    }
  };

And in /api/transcribe I have this:

`import` { NextResponse } from "next/server";

export async function POST(request) {
  // Get formData from request
  const formData = await request.formData();
  // FormData
  /*
  file: File
  model: string
  response_format: string
  propmt: string
  language: string (optional) */

  const apiKEY = process.env.DG_API_KEY;

  if (!apiKEY) {
    return NextResponse.json(
      {
        message:
          "You need to set your API Key as env varibale or with the input.",
      },
      { status: 401, statusText: "Unauthorized" }
    );
  }

  try {
    const response = await fetch(
      "https://api.deepgram.com/v1/listen?model=nova&punctuate=true",
      {
        method: "POST",
        headers: {
          Authorization: `Bearer ${apiKey}`,
        },
        body: formData,
      }
    );
    if (response.ok) {
      const data = await response.json();
      console.log(data); 
      return NextResponse.json(data);
    } else {
      throw new Error("Request failed");
    }
  } catch (error) {
    console.log(error.message);
    return NextResponse.json({ message: "Error" });
  }
}`

My error is this:

![image](https://github.com/deepgram/community/assets/58747332/712f70ea-6d3a-46a0-9e4d-d0aafb09f1ba)

**Comment by lukeocodes** [comment]
Hey Danny!

I'm away from my desk, but I have recently built this Next.js app using the Deepgram Node SDK

https://youtube-demo.deepgram.com

You can find a link to the repo. The Deepgram integration is inside a route.js file as per app routing in Next 13. I'll follow up when I'm back at my desk, hopefully you'll figure it out!

**Reply by DannyPerz** [reply]
Hello Luke,

Thanks for sharing your experience and providing a helpful link! It looks like you've managed to integrate Deepgram into your Next.js application using the Deepgram SDK. That's great.

I will guide myself in your code, to implement it in my application.

Also, if you have any additional tips or specific steps that other developers can follow when integrating Deepgram into Next.js, it would be very helpful to share them.

Thanks again for your answer and I'm looking forward to seeing how you managed to integrate Deepgram into Next.js.

Kind regards.

**Comment by lukeocodes** [comment, marked as answer]
@DannyPerz what version of the SDK are you using? there may have been a typing issue on an earlier version. Updating may resolve this problem

If the problem persists, please raise an issue directly on the SDK, so we can better track known issues against the code

https://github.com/deepgram/deepgram-node-sdk
